app/code/Magento/Theme/Model/PageLayout/Config/Builder.php

```diff
@@ -56,11 +61,14 @@ public function getPageLayoutsConfig()
      */
     protected function getConfigFiles()
     {
-        $configFiles = [];
-        foreach ($this->themeCollection->loadRegisteredThemes() as $theme) {
-            $configFiles = array_merge($configFiles, $this->fileCollector->getFilesContent($theme, 'layouts.xml'));
+        if (empty($this->configFiles)) {
+            $configFiles = [];
```

Comment:
`$configFiles` variable is unnecessary, please use the `configFiles` property straight away.

Reply:
I'm not sure to understand correctly your comment. 😕 

Because putting an `array_merge` instruction directly inside the loop could impact the performances ([source](https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop)). And I'm using a temporary variable to only modify the property once the value is fully calculated.

Reply:
Thanks for the clarification and sorry for the incorrect suggestion, please ignore my previous comment.